sfcasts/field-configurator.md

```diff
@@ -1,189 +1,84 @@
+
+Behind the curtain, the entire purpose of this `Field` object or any of these `Field` classes is to take all of the information we give it and create a `FieldDto`. I'll call `->formatValue()` temporarily and hold "cmd" to jump into that. This actually moved us into a `FieldTrait.php` that field uses.
+
+Check this out! When we call `formatValue()`, what that really does is say `$this->dto->setFormatValueCallable()`. That Dto is the `FieldDto`. So we call nice methods on our `Field` object, but in the background, it's just using all of that information to craft this `FieldDto`. This means that this `FieldDto` contains basically the same information as our `Field` objects here, but the data is going to look different. The methods you call on them will also be a bit different.
+
+All right, let's do our truncating here. Create a private constant: `private const MAX_LENGTH = 25`. Down here, we'll say `if (strlen($field->getFormattedValue()))`, where "formatted value" is just the string that it's about to print out, and then just `return`. Don't make any changes to the field. Just allow the formatted value to be returned like normal. Below, we want to truncate it. So say `$truncatedValue =` and I'm going to use a `u()` function here. Hit "tab" to auto complete that and, just like class, it's added a "use" statement up here. This gives me a nice Unicode string object from the `Form` components.
+
+Inside, I'll pass `$field->getFormattedValue()` and call `->truncate()` with `self:MAX_LENGTH, '...', false`. The last argument just makes this truncate a little cleaner. Oh, and I forgot one of my colons right there. That's better. Now, we're gonna say `$field->setFormattedValue()` and pass it `$truncatedValue`. We're overriding what the formatted value *would* be.
+
+If we go over and refresh... absolutely nothing happens. All of the items in this column *still* have the same length as before. What's happening? We know our configure methods are getting called. So do we have a bug in our code?
+
+Well, when we create a class and make it implement `FieldConfiguratorInterface`, thanks to Symfony's auto configure feature, this gives a special tag to our service called `ea.field_configurator`. It turns out, that's the key to getting your field into the configurator system. You *have* to have a service with that tag.
+
+At your terminal, run:
+
+```terminal
+symfony console debug:container
+```
+
+And we can actually list all the services with that tag by saying:
+
+```terminal
+--tag=ea.field_configurator
+```
```

Comment:
I refactored this to:

```
At your terminal, run `symfony console debug:container`. And we can actually list
all the services with that tag by saying `--tag=ea.field_configurator`:

'''terminal-silent
symfony console debug:container --tag=ea.field_configurator
'''
```

(except that I used `'` instead of the true ` because it's hard to show than in the comment without Github formatting it :) 

Anyways, we ultimately need just ONE terminal block with what I actually typed. Since I sorta half-wrote it, talked and wrote the rest, just put that text inside normal ticks. Then finally, when I'm done, use a `terminal-silent` with the full command.